src/components/modals/CreateAntModal/CreateAntModal.tsx

```diff
@@ -0,0 +1,591 @@
+import Table from 'rc-table';
+import { useEffect, useState } from 'react';
+
+import { useIsMobile, useWalletAddress } from '../../../hooks';
+import { ANTContract } from '../../../services/arweave/AntContract';
+import { useGlobalState } from '../../../state/contexts/GlobalState';
+import {
+  ANTContractJSON,
+  ArweaveTransactionID,
+  ManageAntRow,
+  TRANSACTION_TYPES,
+  VALIDATION_INPUT_TYPES,
+} from '../../../types';
+import { STUB_ARWEAVE_TXID } from '../../../utils/constants';
+import { mapKeyToAttribute } from '../../cards/AntCard/AntCard';
+import { CloseIcon, PencilIcon } from '../../icons';
+import ValidationInput from '../../inputs/text/ValidationInput/ValidationInput';
+import ConfirmAntCreation from '../../layout/ConfirmAntCreation/ConfirmAntCreation';
+import DeployTransaction from '../../layout/DeployTransaction/DeployTransaction';
+import TransactionSuccess from '../../layout/TransactionSuccess/TransactionSuccess';
+import Workflow from '../../layout/Workflow/Workflow';
+
+function CreateAntModal({ show }: { show: boolean }) {
+  const isMobile = useIsMobile();
+  const [{ arweaveDataProvider }, dispatchGlobalState] = useGlobalState();
+  const { walletAddress } = useWalletAddress();
+
+  const [ant, setAnt] = useState<ANTContract>(new ANTContract());
+
+  const [rows, setRows] = useState<[]>([]);
+  const [editingField, setEditingField] = useState<string>();
+  const [modifiedValue, setModifiedValue] = useState<string | number>();
+
+  const [workflowStage, setWorkflowStage] = useState<number>(0);
+
+  const [antContractId, setAntContractId] = useState<
+    ArweaveTransactionID | undefined
+  >();
+  const [isPostingTransaction, setIsPostingTransaction] = useState(false);
+
+  const steps = {
+    1: {
+      title: 'Set ANT Details',
+      status:
+        workflowStage + 1 === 1
+          ? 'pending'
+          : workflowStage + 1 > 1
+          ? 'success'
+          : '',
+    },
+    2: {
+      title: 'Confirm ANT',
+      status:
+        workflowStage + 1 === 2
+          ? 'pending'
+          : workflowStage + 1 > 2
+          ? 'success'
+          : '',
+    },
+    3: {
+      title: 'Deploy ANT',
+      status:
+        workflowStage + 1 === 3
+          ? 'pending'
+          : workflowStage + 1 > 3
+          ? 'success'
+          : '',
+    },
+    4: {
+      title: 'Manage ANT',
+      status: workflowStage + 1 === 4 ? 'success' : '',
+    },
+  };
+  const EDITABLE_FIELDS = [
+    'name',
+    'ticker',
+    'targetID',
+    'controller',
+    'owner',
+    'ttlSeconds',
+  ];
```

Comment:
i believe these are used in `ANTCard` - if so put them in constants

Reply:
slightly different list